QT/timedomain.cpp

```diff
@@ -0,0 +1,107 @@
+#include "timedomain.h"
```

Comment:
we should move the `.cpp` and `.h` files to separate directories

@dudublad

Reply:
They show up in different files in QT, so imo its fine
![image](https://user-images.githubusercontent.com/34686490/143969901-6f22a8e7-ab7d-4f41-9cb8-fa01b36634d1.png)